src/libmeasurement_kit/report/report.cpp

```diff
@@ -44,7 +44,41 @@ void Report::open(Callback<Error> callback) {
     }), callback);
 }
 
-void Report::write_entry(Entry entry, Callback<Error> callback) {
+void Report::write_entry(Entry entry, Callback<Error> callback,
+                         Var<Logger> logger) {
+    if (report_id == "") {
+        auto count = 0;
+        for (auto &reporter : reporters_) {
+            auto rid = reporter->get_report_id();
+            if (rid != "") {
+                report_id = rid;
+                ++count;
+            }
+        }
+        /*
+         * XXX For now we trust the last reporter that provides us with a
+         * non-zero-length report-id. The reason for doing the full scan
+         * over the existing reporters and for asserting is to be ready for
+         * an hypothetical moment where more than one reporter will give
+         * back a good report-id (only ooni_reporter does that).
+         */
+        assert(count <= 1);
```

Comment:
I don't see such a change to easily slip through testing, but sure I'm adding an error for that.